packages/client/src/typed-codecs/typed-codecs.ts

```diff
@@ -0,0 +1,86 @@
+import { type ChainDefinition } from "@/descriptors"
+import {
+  getDynamicBuilder,
+  getLookupFn,
+  MetadataLookup,
+} from "@polkadot-api/metadata-builders"
+import { decAnyMetadata, V14, V15 } from "@polkadot-api/substrate-bindings"
+import { TypedCodecs } from "./types"
+
+export const createConstantEntry = (
+  palletName: string,
+  name: string,
+  lookup: MetadataLookup,
+  { buildConstant }: ReturnType<typeof getDynamicBuilder>,
+) => {
+  const pallet = lookup.metadata.pallets.find((p) => p.name === palletName)
+  const constant = pallet?.constants.find((c) => c.name === name)
+  if (constant == null)
+    throw new Error(`Runtime entry Constant(${palletName}.${name}) not found`)
+  return buildConstant(palletName, name)
+}
+
+const withError =
+  <T>(
+    builder: (pallet: string, name: string) => T,
+    errName: string,
+  ): ((pallet: string, name: string) => T) =>
+  (pallet, name) => {
+    try {
+      return builder(pallet, name)
+    } catch {
+      throw new Error(`Runtime entry ${errName}(${pallet}.${name}) not found`)
+    }
+  }
+
+/**
+ * It provides access the codecs of all possible on-chain interactions.
+ *
+ * @param descriptors  Pass descriptors from `@polkadot-api/descriptors`
+ *                     generated by `papi` CLI.
+ */
+export const getTypedCodecs = async <D extends ChainDefinition>(
+  descriptors: D,
+): Promise<TypedCodecs<D>> => {
+  const metadata = decAnyMetadata(await descriptors.getMetadata()).metadata
+    .value as V14 | V15
+  const target = {}
+  const createProxy = (propCall: (prop: string) => unknown) =>
+    new Proxy(target, {
+      get(_, prop) {
+        return propCall(prop as string)
+      },
+    })
+  const createProxyPath = <T>(pathCall: (a: string, b: string) => T) => {
+    const cache: Record<string, Record<string, T>> = {}
+    return createProxy((a) => {
+      if (!cache[a]) cache[a] = {}
+      return createProxy((b) => {
+        if (!cache[a][b]) cache[a][b] = pathCall(a, b)
+        return cache[a][b]
+      })
+    }) as Record<string, Record<string, T>>
+  }
```

Comment:
As an idea, in here we could use the `has` trap of Proxy to indicate the possible Pallets/entries, as we already have the metadata https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/has

I can't think of a use case though, specially since it will be with the metadata the developer has already downloaded for the descriptors, so let's not overcomplicate it, but I think it's an interesting feature of proxies.